server/src/main/java/com/phoenixoft/teambalanceapp/group/entity/Group.java

```diff
@@ -43,8 +43,9 @@
     @OneToMany(orphanRemoval = true, mappedBy = "group")
     private List<Game> games = new ArrayList<>();
 
-    public boolean removeMember(long userId) {
-        return members.removeIf(member -> member.getId().equals(userId));
+    public boolean removeMember(User member) {
+        member.getGroups().remove(this);
+        return members.removeIf(groupMember -> groupMember.getId().equals(member.getId()));

```

Comment:
This is confusing. Can't understand what's the difference between member and groupMember. I will change it.